pytest/tests/sanity/kickout_offline_validators.py

```diff
@@ -0,0 +1,130 @@
+    """Returns the validator information for the given account_id or None if not found"""
+    for validator in validators:
+        if validator["account_id"] == account_id:
+            return validator
+    return None
+
+
+def assert_block_and_chunk_producer(validator):
+    """Asserts that the given validator information is for a block (and chunk) producer"""
+    assert validator is not None, "Not an active validator"
+    assert validator["num_expected_blocks"] > 0, validator
+    assert validator["num_expected_chunks"] > 0, validator
+    assert validator["num_expected_endorsements"] > 0, validator
+
+
+def assert_chunk_validator_only(validator):
+    """Asserts that the given validator information is for a chunk validator-only validator"""
+    assert validator is not None, "Not an active validator"
+    assert validator["num_expected_blocks"] == 0, validator
+    assert validator["num_expected_chunks"] == 0, validator
+    assert validator["num_expected_endorsements"] > 0, validator
+
+
+def assert_kicked_out(validator, reason):
+    """Asserts that the validator is kicked out for the given reason and zero production"""
+    assert validator is not None, "Not kicked out"
+    assert 'reason' in validator, validator
+    kickout_reasons = validator['reason']
+    assert reason in kickout_reasons, validator
+    kickout_stats = kickout_reasons[reason]
+    assert 'produced' in kickout_stats, validator
+    assert kickout_stats['produced'] == 0
+
+
+def get_epoch_info(node, block_hash):
+    """Returns a tuple of previous kickouts, current validators, and next validators."""
+    block = node.get_block(block_hash)
+    assert 'result' in block, block
+    header = block['result']['header']
+    epoch_id = header['epoch_id']
+    epoch_info = node.get_validators(epoch_id=epoch_id)
+    assert 'result' in epoch_info, epoch_info
+    assert 'prev_epoch_kickout' in epoch_info['result'], epoch_info
+    prev_kickouts = epoch_info['result']['prev_epoch_kickout']
+    assert 'current_validators' in epoch_info['result'], epoch_info
+    current_validators = epoch_info['result']['current_validators']
+    assert 'next_validators' in epoch_info['result'], epoch_info
+    next_validators = epoch_info['result']['next_validators']
+    return (prev_kickouts, current_validators, next_validators)
+
+
+nodes = start_cluster(
+    8, 0, 4, None,
+    [["epoch_length", EPOCH_LENGTH], ["num_block_producer_seats", 8],
+     ["num_chunk_producer_seats", 8], ["num_chunk_validator_seats", 16],
+     ["block_producer_kickout_threshold", KICKOUT_THRESHOLD],
+     ["chunk_producer_kickout_threshold", KICKOUT_THRESHOLD],
+     ["chunk_validator_only_kickout_threshold", KICKOUT_THRESHOLD],
+     ["minimum_validators_per_shard", 1],
+     ["total_supply", "8644440000000000000000000000000000"]] +
+    [["validators", i, "amount", "11000000000000000000000000000000"]
+     for i in range(4)] +
+    [["validators", i, "amount", "110000000000000000000000000000"]
```

Comment:
I suggest to reuse the same constants for numbers, and use `str(11 * 10**...)` to make more obvious whose stake is larger.

Reply:
done